testing/e2e/suites/externaldns_crd.go

```diff
@@ -0,0 +1,330 @@
+package suites
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/Azure/aks-app-routing-operator/api/v1alpha1"
+	"github.com/Azure/aks-app-routing-operator/testing/e2e/infra"
+	"github.com/Azure/aks-app-routing-operator/testing/e2e/logger"
+	"github.com/Azure/aks-app-routing-operator/testing/e2e/manifests"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/rest"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func validExternalDNS() *v1alpha1.ExternalDNS {
+	return &v1alpha1.ExternalDNS{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "test",
+			Namespace: "default",
+		},
+		Spec: v1alpha1.ExternalDNSSpec{
+			TenantID: "123e4567-e89b-12d3-a456-426614174000",
+			DNSZoneResourceIDs: []string{
+				"/subscriptions/123e4567-e89b-12d3-a456-426614174000/resourceGroups/test/providers/Microsoft.network/dnszones/test",
+				"/subscriptions/123e4567-e89b-12d3-a456-426614174000/resourceGroups/test/providers/Microsoft.network/dnszones/test2",
+			},
+			ResourceTypes: []string{"ingress", "gateway"},
+			Identity: v1alpha1.ExternalDNSIdentity{
+				ServiceAccount: "test-sa",
+			},
+		},
+	}
+}
+
+func externalDnsCrdTests(in infra.Provisioned) []test {
+	return []test{
+		{
+			name: "externaldns crd validations",
+			cfgs: builderFromInfra(in).
+				withOsm(in, false, true).
+				withVersions(manifests.OperatorVersionLatest).
+				withZones(manifests.NonZeroDnsZoneCounts, manifests.NonZeroDnsZoneCounts).
+				build(),
+			run: func(ctx context.Context, config *rest.Config, operator manifests.OperatorConfig) error {
+				lgr := logger.FromContext(ctx)
+				lgr.Info("starting test")
+
+				tcs := []struct {
+					name          string
+					ed            *v1alpha1.ExternalDNS
+					expectedError error
+				}{
+					{
+						name:          "valid",
```

Comment:
filters in some tests here as well

Reply:
see above

Reply:
What am I looking for here? I dont see you added Filters to the `ExternalDNSSpec` for any test?

Reply:
Sorry, meant look above on the other comment you left about filters.

Filter doesn't have any restrictions on the value since it's optional, so there's nothing really to test from a CEL perspective. Filter functionality will be covered via integration tests and/or e2e tests that ensure that externaldns only reads gateway resources with the correct filters, and not all that exist in the cluster - but that can really only be tested once the controller for the CRD gets built out

Reply:
When you say externaldns spec - are you referring to the deployment that gets created, or the actual CR?

If the former - yes, we do expect that config to eventually propagate to the deployment args. However, that would happen via the controller that we make for the ExternalDNS CR, but that will be created (and tested) in a separate PR once the CRD definitions are actually in.

If the latter - are you suggesting a UT that sets the filters in an instance of the CR in golang/code and then validates that those filters actually made it to the object that gets saved by the APIserver?

Reply:
More about the later but I see you're not doing a validation of the ExternalDNS resource after upsert

Reply:
I feel like that's more of a test of the Upsert method itself and/or the apiserver's ability to correctly store the object but that will be validated anyway via tests over the controller's functionality, but appreciate the callout + will make sure that there are adequate tests once we get the controller up